### ISSUE TITLE
Sharing central folder

### ISSUE THREAD

**Issue by stevenbird** [aikuma/aikuma#431]
@hleeldc proposes that we implement file sharing update (#371) by sharing the central Google drive with all Aikuma users. The Aikuma client would keep track of which items it was following (a per-device file) and this would be used for querying the shared central drive (e.g. full-content query of json files for the item id, then retrieve all files from the shared central drive to the private google drive). Any item followed by any device owned by a userid would end up synced on all devices owned by that userid.

This solution avoids the need to store following information on the central server, and avoids the need to do queries on the central server.

**Comment by Lingomat** [comment]
Suppose we lose a phone and replace it. How do we reestablish an appropriate set of files to restore to a replacement phone? I take from your solution that it would be necessary to copy Aikuma's 'following' file from another handset. If that happened transparently, because, say, we're the same userid and because we have a peer-to-peer network established, then I'm not seeing a drawback.

**Comment by stevenbird** [comment]
@hleeldc would you please confirm that it is possible to do these steps (or suggest another way)?

1. full-text search for all files containing `item_id="ualwfcixlzpu"`
2. returns a list of google file identifiers for the json files containing this item
3. get the filename from the google file identifier
4. check that the file has not already been obtained
5. construct the full .wav and -mapping.json filenames, and get the google file identifiers
6. copy into private google drive from the Incoming/aikuma shared folder

Note that when doing search via the Google Drive web interface, we get matches on filenames and on contents (even when searching for substrings?) Hopefully in step 1 above we can constrain the matching to filenames or contents.

**Comment by stevenbird** [comment]
For efficient query, we will store Aikuma's file-level metadata in the `aikuma_metadata` property which is a string containing keywords like `speaker_JFELALELFIXAL iso639_kor`.

**Comment by stevenbird** [comment]
@hleeldc do we also need to have `item_id-ualwfcixlzpu` for all files (including metadata files) that are for item ualwfcixlzpu?

**Comment by stevenbird** [comment]
@hleeldc user_id1 can create a derivative file (respeaking or translation), e.g. `ualwfcixlzpu-testieoda@gmail.com-interpret-839273.wav`. Suppose this is tagged by user_id2, represented in an empty file `ualwfcixlzpu-testieoda@gmail.com-interpret-839273-custom-i_like_this`, stored under user_id2. In order to find all tags for a given source file or derivative file, @lisaslyis points out that we need to store the source or derivative file's filename (or basename) as a property, e.g.:

```
file-ualwfcixlzpu-testieoda@gmail.com-interpret-839273
```

**Comment by stevenbird** [comment]
The previous comment concerns searching the private and shared Google drive, so it only happens when Aikuma has network access. Aikuma also needs to be able to search it's local storage for tags.

**Comment by stevenbird** [comment]
For example, when storing the following file in the Google drive:

```
/aikuma/v02/testslfle@gmail_com/items/ualwfcixlzpu/ualwfcixlzpu-testslfle@gmail.com-respeak-829120.wav
```

We add the property: `ualwfcixlzpu/ualwfcixlzpu-testslfle@gmail.com-respeak-829120`

**Comment by stevenbird** [comment]
Note that all derivative files should have a property which tells us the source id. This enables us to find all files that concern a given source (e.g. for updating, cf #434).

**Comment by stevenbird** [comment]
These file properties are set when the file is created: `item_id, user_id, file_type, version, name`.

Properties based on tags (languages, speakers, olac-discourse-type, custom-tag) need to be updated each time a new tag file is added.

**Comment by stevenbird** [comment]
@hleeldc – please confirm that this is resolved in #491.

**Comment by stevenbird** [comment]
@lisaslyis – would you please confirm that you are setting the file properties when a file is created: `item_id, user_id, file_type, version, name`.

**Comment by stevenbird** [comment]
Resolved by #542
